API.md

```diff
@@ -0,0 +1,235 @@
+A space-separated list of the types or modes of tests to run. Aliased as `-m`. Available modes:
+
+- `type` - Runs Typescript type-checking
+- `lint` - Runs ESLint
+- `unit` - Runs Jest-based unit tests
+
+Defaults to all modes. 
+
+This will initialize the `"start"` and `"test"` scripts in the `package.json` to pass the matching argument.
+
+### Manual installation
+
+`benmvp create` will automatically add the latest version of `@benmvp/cli` as a dev dependency to your library, even if a `package.json` exists. However, you can manually install it.
+
+Install via [Yarn](https://yarnpkg.com/lang/en/docs/managing-dependencies/):
+
+```sh
+yarn add --dev @benmvp/cli
+```
+
+Install via [NPM](https://docs.npmjs.com/getting-started/installing-npm-packages-locally):
+
+```sh
+npm install --save-dev @benmvp/cli
+```
+
+You will want to create scripts to call [`benmvp test`](#benmvp-test), [`benmvp start`](#benmvp-start), and [`benmvp build`](#benmvp-build) for ease of use.
+
+## `benmvp test`
+
+Runs a one-time pass of typing, linting & unit tests for the library.
+
+### Examples
+
+To run all modes (default behavior):
+
+```sh
+benmvp test
+```
+
+To run just unit tests:
+
+```sh
+benmvp test --modes lint
+```
+
+To run typing & linting:
+
+```sh
+benmvp start -m type lint
+```
+
+### Arguments
+
+#### `--modes`
+
+A space-separated list of the types or modes of tests to run. Aliased as `-m`. Available modes:
+
+- `type` - Runs Typescript type-checking
+- `lint` - Runs ESLint
+- `unit` - Runs Jest-based unit tests
+
+Defaults to all modes. 
+
+## `benmvp start`
+
+Runs the lib's tests in on-going watch mode during active development. When building a library, the best way to validate that it is working is by running tests. Having the tests running in watch mode make it easier to ensure that refactors are non-breaking.
+
+### Examples
+
+To run all modes (default behavior):
+
+```sh
+benmvp start
+```
+
+To run just type-checking:
+
+```sh
+benmvp start --modes type
+```
+
+To run linting & unit test:
+
+```sh
+benmvp start -m lint unit
+```
+
+### Arguments
+
+#### `--modes`
+
+A space-separated list of the types or modes of tests to run. Aliased as `-m`. Available modes:
+
+- `type` - Runs Typescript type-checking
+- `lint` - Runs ESLint
+- `unit` - Runs Jest-based unit tests
+
+Defaults to all modes.
+
+## `benmvp build`
+
+Builds the library into desired module formats at the specified location.
+
+### Examples
+
+To specify an alternate output directory:
+
+```sh
+benmvp build --output-path ./built
+```
+
+NOTE: You will need to manually update the default `"main"`, `"module"`, `"jsnext:main"`, `"browser"`, and `"types"` properties in your `package.json` to reflect the new location of the built module formats.
+
+To exclude bundled web distribution format:
+
+```sh
+benmvp build --formats esm umd type
+```
+
+To put ESM & web distribution formats in an alternate build location with continuous watch:
+
+```sh
+benmvp build -f esm dist -o ./built -w
```

Comment:
Inconsistency: you use `--formats` in other places, but `-f` here. While I like showing off the alias I think it would be better to use full names in all place and do so consistently.

Reply:
Yeah you're right. Was trying to show off the aliases, but I agree